src/python/pants/backend/jsx/target_types.py

```diff
@@ -74,3 +81,74 @@ class JSXSourcesGeneratorTarget(TargetFilesGenerator):
     copied_fields = COMMON_TARGET_FIELDS
     moved_fields = (JSXDependenciesField,)
     help = "Generate a `jsx_source` target for each file in the `sources` field."
+
+
+class JSXTestDependenciesField(JSXDependenciesField):
+    pass
+
+
+class JSXTestSourceField(JSXSourceField, JSTestRuntimeSourceField):
+    expected_file_extensions = JSX_FILE_EXTENSIONS
+
+
+class JSXTestTimeoutField(TestTimeoutField):
+    pass
+
+
+class JSXTestExtraEnvVarsField(TestExtraEnvVarsField):
+    pass
```

Comment:
I'm honestly not sure why I declare sublcasses for each field, but it seems prevalent in all backends.